File: Technology Transfer _ Sulambi VMS/Source Code/sulambi-backend-main/sulambi-backend-main/app/routes/reports.py
from flask import Blueprint, request
from ..controllers import reports
from ..middlewares.requiredParams import reportsParams

ReportsBlueprint = Blueprint('reports', __name__, url_prefix="/reports")

@ReportsBlueprint.get("/")
def getAllReportDetails():
  return reports.getAllReports()

@ReportsBlueprint.get("/analytics/external/<eventId>")
def getExternalReportDetails(eventId):
  return reports.getReportCalculations(eventId, "external")

@ReportsBlueprint.get("/analytics/internal/<eventId>")
def getInternalReportDetails(eventId):
  return reports.getReportCalculations(eventId, "internal")

@ReportsBlueprint.post("/external/<eventId>")
def createExternalReportRoute(eventId):
  return reports.createReport(eventId, "external")

@ReportsBlueprint.post("/internal/<eventId>")
def createInternalReportRoute(eventId):
  return reports.createReport(eventId, "internal")

@ReportsBlueprint.delete("/external/<reportId>")
def deleteExternalReportRoute(reportId):
  return reports.deleteReport(int(reportId), "external")

@ReportsBlueprint.delete("/internal/<reportId>")
def deleteInternalReportRoute(reportId):
  return reports.deleteReport(int(reportId), "internal")

@ReportsBlueprint.before_request
def reportsMiddleware():
  if (request.method != "OPTIONS"):
    if (request.method not in ["GET", "DELETE", "PATCH"]):
      missingParams = None

      # external report creation
      if (("/api/reports/external" in request.path) and request.method == "POST" and request.view_args.get("eventId") != None):
        missingParams = reportsParams.createExternalReportParamCheck()

      # internal report creation
      elif (("/api/reports/internal" in request.path) and request.method == "POST" and request.view_args.get("eventId") != None):
        missingParams = reportsParams.createInternalReportParamCheck()

      if (missingParams != None):
        return missingParams

File: Technology Transfer _ Sulambi VMS/Source Code/sulambi-frontend-main/sulambi-frontend-main/src/api/reports.ts
import axios from "./init";

const basePath = "/reports";

export const getAllReports = () => {
  return axios.get(`${basePath}/`);
};

export const getReportAnalytics = (
  eventId: number,
  type: "external" | "internal"
) => {
  return axios.get(`${basePath}/analytics/${type}/${eventId}`);
};

export const createReport = (
  eventId: number,
  type: "external" | "internal",
  reportData: FormData
) => {
  return axios.post(`${basePath}/${type}/${eventId}`, reportData, {
    headers: {
      "Content-Type": "multipart/form-data",
    },
  });
};

export const deleteReport = (
  reportId: number,
  type: "external" | "internal"
) => {
  return axios.delete(`${basePath}/${type}/${reportId}`);
};


